apps/extension/src/extension/utils.ts

```diff
@@ -0,0 +1,31 @@
+import browser from "webextension-polyfill";
+import { Env, MessageSender, KVPrefix } from "../router/types";
+import { ExtensionKVStore } from "@anoma/storage";
+
+const ROUTER_ID_KEY = "anomaExtensionRouterId";
+const store = new ExtensionKVStore(KVPrefix.LocalStorage);
+
+export const getAnomaRouterId = async (): Promise<number | undefined> => {
+  const storedId = await store.get(ROUTER_ID_KEY);
+  if (!storedId) {
+    const id = Math.floor(Math.random() * 1000000);
+    await store.set(ROUTER_ID_KEY, id);
+  }
+  return store.get(ROUTER_ID_KEY);
+};
+
+// Determine if content-scripts can be executed in this environment
+export class ContentScriptEnv {
+  static readonly produceEnv = (sender: MessageSender): Env => {
+    const isInternalMsg = sender.id === browser.runtime.id;
+
+    return {
+      isInternalMsg,
+      requestInteraction: () => {
```

Comment:
What is `requestInteraction` used for? I see that we use Utility types in several occasions to remove this from `Env`. But I do not really see what what we use it for 🤔

Reply:
That's going to be important later when we're requesting connections to the extension from the interface.